service/matching/taskWriter.go

```diff
@@ -156,6 +156,9 @@ func (w *taskWriter) allocTaskIDs(count int) ([]int64, error) {
 		if w.taskIDBlock.start > w.taskIDBlock.end {
 			// we ran out of current allocation block
 			newBlock, err := w.tlMgr.allocTaskIDBlock(w.taskIDBlock.end)
+			if err != nil && w.tlMgr.errShouldUnload(err) {
```

Comment:
what about merging the error checking logic to within `errShouldUnload`?

Reply:
errShouldUnload will already behave correctly if err is nil. I left the check here to help the reader of the code. I.e. so someone debugging this function does not have to dive into errShouldUnload to see if it does the right thing with nil.